Engines/Wine/QuickScript/Installer Script/script.js

```diff
@@ -32,14 +32,26 @@ module.default = class InstallerScript extends QuickScript {
 
         // let user select wine settings if desired
         if (this._wineUserSettings) {
-            const architectures = ["x86", "amd64"];
-            const selectedArchitecture = setupWizard.menu(
-                tr("Please select the wine architecture."),
-                ["x86 (recommended)", "amd64"],
-                "x86 (recommended)"
-            );
+            if (operatingSystemFetcher.fetchCurrentOperationSystem().getFullName() == "Mac OS X") {
+                const architectures = ["x86", "amd64", "x86on64"];
+                const selectedArchitecture = setupWizard.menu(
+                    tr("Please select the wine architecture."),
+                    ["x86 (recommended)", "amd64", "x86 on amd64 (beta)"],
```

Comment:
Mac OS X does have both x86 and x86 on amd64? This will probably be confusing for many users.

Reply:
Yep, that's the point... x86on64 runs x86 programs with 64bits binaries

Reply:
I don't see how we could "hide" this complexity in fact

Reply:
Ok. My understanding until now was basically that x86 does not work anymore. I didn't really follow the OSX stuff though.

Reply:
It does not work on 10.15 but works well on 10.14-
In return, it has all versions without needing a complex patch

Reply:
Maybe there should be note regarding >= 10.15 and < 10.14. It's not possible to do this magic automatically?

Reply:
This is the way it should be done yep

Reply:
Like this? https://coderwall.com/p/4yz8dq/determine-os-x-version-from-the-command-line